test/GPv2Settlement/ComputeTradeExecutions.t.sol

```diff
@@ -0,0 +1,192 @@
+            kind: bytes32(0),
+            partiallyFillable: false,
+            sellTokenBalance: GPv2Order.BALANCE_ERC20,
+            buyTokenBalance: GPv2Order.BALANCE_ERC20
+        });
+    }
+
+    function setTokenPrices(uint256 _sellPrice, uint256 _buyPrice) internal {
+        IERC20[] memory tokens = new IERC20[](2);
+        tokens[0] = sellToken;
+        tokens[1] = buyToken;
+        uint256[] memory prices = new uint256[](2);
+        prices[0] = _sellPrice;
+        prices[1] = _buyPrice;
+        encoder.tokenRegistry.tokenRegistry().setPrices(tokens, prices);
+    }
+
+    function computeSettlementForOrder(GPv2Order.Data memory order)
+        internal
+        returns (uint256 executedSellAmount, uint256 executedBuyAmount)
+    {
+        setTokenPrices(sellPrice, buyPrice);
+        encoder.signEncodeTrade(vm, trader, order, domainSeparator, GPv2Signing.Scheme.Eip712, executedAmount);
+
+        SettlementEncoder.EncodedSettlement memory encoded = encoder.encode(settlement);
+        (GPv2Transfer.Data[] memory inTransfers, GPv2Transfer.Data[] memory outTransfers) =
+            settlement.computeTradeExecutionsTest(encoded.tokens, encoded.clearingPrices, encoded.trades);
+
+        // TODO: this is required due some weirdness whereby if `settlement.computeTradeExecutionsTest` reverts,
+        // for some reason it continues executing here and results in a panic with an index out of bounds error.
+        if (inTransfers.length == 1 && outTransfers.length == 1) {
+            executedSellAmount = inTransfers[0].amount;
+            executedBuyAmount = outTransfers[0].amount;
+        }
+    }
+}
+
+contract ComputeTradeExecutions is BaseComputeTradeExecutions {
+    using SettlementEncoder for SettlementEncoder.State;
+    using Order for GPv2Order.Data;
+
+    function test_should_not_allocate_additional_memory() public {
+        assertTrue(settlement.computeTradeExecutionMemoryTest() == 0);
+    }
+
+    function test_should_compute_in_out_transfers_for_multiple_trades() public {
+        uint256 tradeCount = 10;
+        for (uint256 i = 0; i < tradeCount;) {
+            GPv2Order.Data memory order = partialOrder();
+            order.kind = GPv2Order.KIND_BUY;
+            order.partiallyFillable = true;
+            encoder.signEncodeTrade(vm, trader, order, domainSeparator, GPv2Signing.Scheme.Eip712, 0.7734 ether);
+
+            unchecked {
+                ++i;
+            }
+        }
+
+        setTokenPrices(sellPrice, buyPrice);
+        SettlementEncoder.EncodedSettlement memory encoded = encoder.encode(settlement);
+        (GPv2Transfer.Data[] memory inTransfer, GPv2Transfer.Data[] memory outTransfer) =
+            settlement.computeTradeExecutionsTest(encoded.tokens, encoded.clearingPrices, encoded.trades);
+
+        assertEq(inTransfer.length, tradeCount);
+        assertEq(outTransfer.length, tradeCount);
+    }
+
+    function test_revert_if_the_order_is_expired() public {
+        vm.warp(42);
+
```

Comment:
This should be unneeded, right?
```suggestion
```

Reply:
When checking the `block.timestamp` at this point, it yields `1`. Given that a `validTo` of `0` is indistinguishable to an default Solidity initialised order with a `validTo` of `0`, I elected to set an explicit `block.timestamp` to yield a more readable test, and enforce a difference to a non-initialised Solidity struct for `validTo`.

Reply:
Good point! Maybe we can add this as an inline comment?